internal/stores/metadata.go

```diff
@@ -422,6 +419,17 @@ func (s *SQLStore) AncestorContracts(ctx context.Context, id types.FileContractI
 	return contracts, nil
 }
 
+func (s *SQLStore) ArchiveContracts(ctx context.Context, ids []types.FileContractID, reason string) error {
+	cs, err := contracts(s.db, ids)
+	if err != nil {
+		return err
+	}
+
+	return s.retryTransaction(func(tx *gorm.DB) error {
+		return archiveContracts(tx, cs, reason)
+	})
```

Comment:
Doesn't this need to update `knownContracts`?

Reply:
I figured it wouldn't because of the potential race condition?

Reply:
Hm I guess that's right. If we don't delete contracts anymore they will just stay known forever. Which isn't bad.